plugins/c8y_remote_access_plugin/src/input.rs

```diff
@@ -0,0 +1,157 @@
+use clap::ArgGroup;
+use clap::Parser;
+use miette::ensure;
+use miette::miette;
+use miette::Context;
+use serde::Deserialize;
+
+use crate::csv::deserialize_csv_record;
+
+#[derive(Parser, Deserialize, Debug, PartialEq, Eq)]
+pub struct RemoteAccessConnect {
+    device_id: String,
+    host: String,
+    port: u16,
+    key: String,
+}
+
+#[derive(Parser)]
+#[clap(group(ArgGroup::new("install").args(&["init", "cleanup", "connect_string"])))]
+#[clap(
+name = clap::crate_name!(),
+version = clap::crate_version!(),
+about = clap::crate_description!(),
+)]
+struct Cli {
+    #[arg(long)]
+    /// Complete the installation of c8y-configuration-plugin by declaring the supported operation.
+    init: bool,
+
+    #[arg(long)]
+    /// Clean up c8y-configuration-plugin, deleting the supported operation from tedge.
+    cleanup: bool,
+
+    /// The SmartREST connect message, forwarded from mosquitto by tedge-mapper.
+    ///
+    /// Can only be provided when neither '--init' nor '--cleanup' are provided.
+    connect_string: Option<String>,
+}
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum Command {
```

Comment:
It would have been nice if clap could parse the input arguments to this enum directly as that would eliminate the need for us to validate exclusive arg combinations.

Reply:
Yeah, I don't think this is possible unfortunately. The `#[clap(group(...))]` does at least give good error messages when multiple arguments are provided.